### ISSUE TITLE
Lighthouse warns when an anchor tag doesn't have a `href` attribute and does contain an image tag.

### ISSUE THREAD

**Issue by romainmenke** [GoogleChrome/lighthouse#15820]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

http://www.romainmenke.com/lighthouse-bug-anchors-without-href-attribute/without.html

### What happened?

> Links are not crawlable

<img width="756" alt="Screenshot 2024-02-21 at 19 24 27" src="https://github.com/GoogleChrome/lighthouse/assets/11521496/69ba2cb1-4388-4b3f-a522-9d2f82c6227f">

There are a bunch of warnings on that page but only `Links are not crawlable` is relevant to this issue.

### What did you expect?

The [HTML specification](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) clearly envisions a dual mode for anchor elements.

Either they have an `href` attribute, which makes them true anchor elements or they do not, in which case they represent a placeholder.

This concept is also used in CSS for the location pseudo classes : https://drafts.csswg.org/selectors/#the-any-link-pseudo

These only match anchors that do have an `href` attribute.

-------

When templating (server-side) it really helps to be able to wrap a bunch of html with an anchor tag and optionally give it an `href` attribute only when it needs to be a link.

This makes the variable part in the template much smaller (only 1 attribute).

-------

I expect Lighthouse to embrace the relevant specifications and to not give any warnings here.

### What have you tried?

N/A

### How were you running Lighthouse?

Chrome DevTools

### Lighthouse Version

latest

### Chrome Version

latest

### Node Version

N/A

### OS

N/A

### Relevant log output

```shell
N/A
```

**Comment by adamraine** [comment]
> I expect Lighthouse to embrace the relevant specifications and to not give any warnings here.

Yeah I think we can discuss changing this. Although the link is not crawlable this doesn't appear to be a bad practice that search documentation warns against.

> I also expect Lighthouse to have consistent behavior between anchor tags without href when they contain text vs. when they contain images.

Not sure what you are referring to here, what differences did you noticed between anchor tags containing images vs text?

**Comment by romainmenke** [comment]
> > I also expect Lighthouse to have consistent behavior between anchor tags without href when they contain text vs. when they contain images.
>
> Not sure what you are referring to here, what differences did you noticed between anchor tags containing images vs text?


This triggers a warning:

```html
<a>
	<img src="https://picsum.photos/200/300" alt="" width="200" height="300" loading="lazy">
</a>
```

This does not:

```html
<a>
	Hello
</a>
```

**Comment by adamraine** [comment]
> ```html
> <a>
> 	Hello
> </a>
> ```

The audit fails for me here as well. Not noticing any difference between images/text.

**Comment by romainmenke** [comment]
You are right!

I was testing with an anchor element that also had an `id` attribute.
My assumption that it had to do with image/text content was incorrect.

**Comment by romainmenke** [comment]
Thank you for updating the title.
I've also edited the initial report and the html content of the [example page](https://www.romainmenke.com/lighthouse-bug-anchors-without-href-attribute/without.html)

This better reflects the issue.

**Comment by hiroya-uga** [comment]
This is a very interesting discussion.

I agree that pointing out `<a>` elements without an `href` attribute is valuable.
On the other hand, the placeholder role that `a:not([href])` can serve is also significant from an HTML perspective.

Looking at the already closed PR #15894, it seems too complex to resolve quickly.
As a potential solution, how about excluding cases where the `aria-current` attribute from WAI-ARIA is present?

Because, `a:not([href])[aria-current="page"]` clearly indicates a placeholder use case, so I believe it doesn't need to be flagged as an error.

Thank you for reading!

related specs:

- [the a element - HTML Living Standard](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:~:text=If%20a%20site%20uses%20a%20consistent%20navigation%20toolbar%20on%20every%20page%2C%20then%20the%20link%20that%20would%20normally%20link%20to%20the%20page%20itself%20could%20be%20marked%20up%20using%20an%20a%20element%3A)
- [a without href - ARIA in HTML](https://www.w3.org/TR/html-aria/#:~:text=href%20attribute.-,a%20without%20href,-role%3Dgeneric)
- [aria-current - WAI-ARIA 1.3](https://www.w3.org/TR/wai-aria-1.3/#aria-current)
